src/engine.rs

```diff
@@ -298,7 +299,10 @@ impl Engine<DefaultFileBuilder, FilePipeLog<DefaultFileBuilder>> {
         )
     }
 
-    pub fn dump(path: &Path, raft_groups: &[u64]) -> Result<Vec<LogItem>> {
+    pub fn dump<'a>(
+        path: &Path,
+        raft_groups: &'a [u64],
```

Comment:
Actually, we can remove this parameter, and do the filtering in "ctl". So that you don't need to store it in iterator anymore.

Reply:
@tabokie If we remove this parameter, method `dump` seems do nothing but convert `LogItemReader.next()` to a normal  iterator, is that OK?

Reply:
Yes, ctl still has this parameter.

Reply:
👌